tests/unit/llms/providers/test_bam.py

```diff
@@ -36,3 +36,68 @@ def test_basic_interface(provider_config):
     llm = bam.load()
     assert isinstance(llm, LangChainInterface)
     assert bam.default_params
+
+
+def test_params_handling(provider_config):
+    """Test that not allowed parameters are removed before model init."""
+    config.init_empty_config()  # needed for checking the config.dev_config.llm_params
+
+    # first three parameters should be removed before model init
+    # rest need to stay
+    params = {
+        "unknown_parameter": "foo",
+        "min_new_tokens": 1,
+        "max_new_tokens": 10,
+        "temperature": 0.3,
+        "verbose": True,
+    }
+
+    bam = BAM(model="uber-model", params=params, provider_config=provider_config)
+    llm = bam.load()
+    assert isinstance(llm, LangChainInterface)
+    assert bam.default_params
+    assert bam.params
+
+    # known parameters should be there
+    assert "min_new_tokens" in bam.params
+    assert bam.params["min_new_tokens"] == 1
+
+    assert "max_new_tokens" in bam.params
+    assert bam.params["max_new_tokens"] == 10
+
+    # unknown parameters should be filtered out
+    assert "unknown_parameter" not in bam.params
+
+
+def test_params_handling_none_values(provider_config):
+    """Test handling parameters with None values."""
+    config.init_empty_config()  # needed for checking the config.dev_config.llm_params
+
+    # first three parameters should be removed before model init
+    # rest need to stay
```

Comment:
same comment here....  comment doesn't align w/ the test logic

Reply:
fixed @bparees